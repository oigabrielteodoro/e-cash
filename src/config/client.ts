export const baseURL = import.meta.env.VITE_API_URL as string
