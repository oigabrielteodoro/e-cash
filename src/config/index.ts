export * from './theme'
export * from './client'
