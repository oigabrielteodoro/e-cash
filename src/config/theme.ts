export const theme = {
  grid: {
    sideBar: {
      open: '25rem',
      closed: '10rem',
    },
  },
  breakpoints: {
    xsm: '320px',
    sm: '425px',
    md: '768px',
    lg: '1440px',
    xlg: '1920px',
    greaterLg: '1441px',
  },
  radius: {
    card: '0.5rem',
    popover: '1rem',
    modal: '1rem',
  },
  font: {
    family: 'Inter',
    sizes: {
      disclaimer: '1.2rem',
      small: '1.4rem',
      paragraph: '1.6rem',
      subtitle: '2.2rem',
      title: '3.2rem',
    },
  },
  layers: {
    base: 1,
    alwaysOnTop: 2,
  },
  colors: {
    white: '#ffffff',
    green: {
      100: '#ecf7ee',
      200: '#c8e2cd',
      500: '#5fb263',
    },
    blue: {
      100: '#e7eff9',
      200: '#b9cfe6',
      300: '#a4b7eA',
      500: '#3a67e1',
      900: '#060c30',
    },
    red: {
      100: '#faede9',
      200: '#eccbc0',
      400: '#fc6471',
      500: '#db5a3a',
    },
    yellow: {
      100: '#fef8eb',
      200: '#f4e1c0',
      500: '#e29936',
    },
    neutral: {
      100: '#eef1f5',
      200: '#f5f5f5',
      300: '#cfd7e3',
      400: '#afbdd0',
      500: '#a1a7ae',
      600: '#4d4f51',
      700: '#393e48',
      900: '#18171d',
    },
  },
  shadow: {
    blue: {
      300: '#a4b7ea30',
    },
    red: {
      200: '#eccbc030',
      400: '#fc647120',
    },
    neutral: {
      500: '#a1a7ae20',
    },
  },
  dropShadow: {
    popover: '0 0 10rem 1rem rgba(0, 0, 0, 0.1)',
  },
}
