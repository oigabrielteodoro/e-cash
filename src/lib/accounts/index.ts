export const agencyNumberWithoutDigitMask = '9999'
export const accountNumberWithDigitMask = '999999-9'
