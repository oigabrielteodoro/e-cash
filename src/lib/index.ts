export * from './matcher'
export * from './router'
export * from './formatters'
export * from './transformers'
export * from './input'
export * from './sidebar'
export * from './accounts'
export * from './masker'
export * from './text'
export * from './id'
export * from './email'
export * from './password'
export * from './url'
