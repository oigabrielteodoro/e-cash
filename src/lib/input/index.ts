export * from './useInput'
export * from './getValidationErrors'
