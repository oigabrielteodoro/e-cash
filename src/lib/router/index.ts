export * from './Router'

export const SIGN_IN = '/'
export const DASHBOARD = '/dashboard'
