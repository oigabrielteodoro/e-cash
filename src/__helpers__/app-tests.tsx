import React, { useEffect, ReactElement, ReactNode } from 'react'
import { render as rtlRender } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import {
  DefaultOptions,
  QueryCache,
  QueryClient,
  QueryClientProvider,
} from 'react-query'
import { MemoryRouter, Routes, Route } from 'react-router-dom'
import { generateId } from 'lib'
import nock from 'nock'

import { pipe } from 'fp-ts/function'
import { TaskEither, map, mapLeft } from 'fp-ts/TaskEither'

import { AppLayout, NotificationContainer } from 'ui'
import { clearToken, queryConfigDefault, setToken } from 'client'

type Options = {
  initialRoute?: string
  initialRoutePath?: string
  authenticated?: boolean
  queryClient?: QueryClient
  routePaths?: string[]
}

type Props = {
  children?: ReactNode
  client?: QueryClient
}

type Callback = (result: unknown) => unknown
type MapAll = (
  fn: Callback,
) => (data: TaskEither<unknown, unknown>) => TaskEither<unknown, unknown>

const queryCache = new QueryCache()
const defaultQueryClient = new QueryClient({
  defaultOptions: queryConfigDefault as DefaultOptions,
  queryCache: queryCache,
})

export function ReactQueryWrapper({
  children,
  client = defaultQueryClient,
}: Props) {
  return <QueryClientProvider client={client}>{children}</QueryClientProvider>
}

export function ReactRouterWrapper({
  initialRoute = '/',
  children,
  routePaths,
  authenticated = true,
}: Options & Pick<Props, 'children'>) {
  function MockedPage({ title }: { title: string }) {
    if (authenticated) {
      return (
        <AppLayout>
          <AppLayout.Content>Page {title}</AppLayout.Content>
        </AppLayout>
      )
    }

    return <h1>Page {title}</h1>
  }

  return (
    <MemoryRouter initialEntries={[initialRoute]}>
      <Routes>
        <Route path={initialRoute} element={children} />
        {routePaths?.map((routePath) => (
          <Route
            key={routePath}
            path={routePath}
            element={<MockedPage title={routePath} />}
          />
        ))}
      </Routes>
    </MemoryRouter>
  )
}

export function ReactHookWrapper({
  initialRoute = '/',
  children,
  routePaths,
  authenticated = true,
  client = defaultQueryClient,
}: Options & Props) {
  return (
    <ReactQueryWrapper client={client}>
      <ReactRouterWrapper
        initialRoute={initialRoute}
        routePaths={routePaths}
        authenticated={authenticated}
      >
        {children}
      </ReactRouterWrapper>
    </ReactQueryWrapper>
  )
}

function render(
  ui: ReactElement,
  {
    initialRoute = '/',
    queryClient = defaultQueryClient,
    authenticated = true,
    routePaths,
  }: Options = {},
) {
  function Wrapper({ children }: Props) {
    useEffect(() => {
      if (authenticated) {
        setToken(generateId())

        return () => {
          clearToken()
        }
      }
    }, [])

    return (
      <ReactQueryWrapper client={queryClient}>
        <ReactRouterWrapper
          initialRoute={initialRoute}
          routePaths={routePaths}
          authenticated={authenticated}
        >
          {children}
        </ReactRouterWrapper>

        <NotificationContainer />
      </ReactQueryWrapper>
    )
  }

  return {
    ...rtlRender(ui, {
      wrapper: Wrapper,
    }),
  }
}

export function unsafe<T>(value: unknown) {
  return value as T
}

export const mapAll: MapAll = (fn) => (data) => {
  return pipe(data, map(fn), mapLeft(fn))
}

export function getErrorMessage(errors: unknown): string {
  return Array.isArray(errors) ? errors[0].message : ''
}

export * from '@testing-library/react'
export {
  render,
  userEvent,
  nock,
  defaultQueryClient as queryClient,
  queryCache,
}
