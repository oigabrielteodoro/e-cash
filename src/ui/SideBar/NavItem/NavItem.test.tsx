/* eslint-disable react/display-name */
import React from 'react'
import { AiOutlineDashboard, AiOutlineUser } from 'react-icons/ai'

import { render, screen, userEvent } from '__helpers__/app-tests'
import { theme } from 'config'
import { sidebarStore } from 'lib'

import { NavItem } from '.'

function MockedComponent() {
  return (
    <>
      <NavItem to='/dashboard' icon={AiOutlineDashboard}>
        Dashboard
      </NavItem>
      <NavItem to='/transactions' icon={AiOutlineUser}>
        Transactions
      </NavItem>
    </>
  )
}

describe('NavItem', () => {
  it('should be able match snapshot', () => {
    const { container } = render(
      <NavItem to='/' icon={AiOutlineDashboard}>
        Link
      </NavItem>,
    )

    expect(container).toMatchSnapshot()
  })

  it('should be able render correctly', async () => {
    render(
      <NavItem to='/' icon={AiOutlineDashboard}>
        Link
      </NavItem>,
    )

    expect(screen.getByLabelText('link')).toBeInTheDocument()
  })

  it('should be able render with active state', async () => {
    render(
      <NavItem to='/' icon={AiOutlineDashboard}>
        Dashboard
      </NavItem>,
    )

    expect(screen.getByLabelText('dashboard')).toBeInTheDocument()
    expect(screen.getByLabelText('dashboard icon')).toHaveStyle(
      `background: ${theme.colors.white}`,
    )
  })

  it('should be able render with active and open state', async () => {
    sidebarStore.setState({
      isOpen: true,
    })

    render(
      <NavItem to='/' icon={AiOutlineDashboard}>
        Dashboard
      </NavItem>,
    )

    expect(screen.getByLabelText('dashboard')).toBeInTheDocument()
    expect(screen.getByText('Dashboard')).toBeInTheDocument()
    expect(screen.getByLabelText('dashboard icon')).toHaveStyle(
      `background: ${theme.colors.white}`,
    )
  })

  it('should not be able render with active state when his is not active', async () => {
    render(
      <NavItem to='/wrong-url-to-active-state' icon={AiOutlineDashboard}>
        Dashboard
      </NavItem>,
    )

    expect(screen.getByLabelText('dashboard')).toBeInTheDocument()
    expect(screen.getByLabelText('dashboard icon')).not.toHaveStyle(
      `background: ${theme.colors.white}`,
    )
  })

  it('should be able navigate to transactions page', async () => {
    render(<MockedComponent />, {
      initialRoute: '/dashboard',
      initialRoutePath: '/dashboard',
      routePaths: ['/transactions'],
    })

    expect(screen.getByLabelText('dashboard')).toBeInTheDocument()
    expect(screen.getByLabelText('transactions')).toBeInTheDocument()

    expect(screen.getByLabelText('dashboard icon')).toHaveStyle(
      `background: ${theme.colors.white}`,
    )

    const link = screen.getByLabelText('transactions')

    userEvent.click(link)

    expect(await screen.findByLabelText('dashboard')).toBeInTheDocument()
    expect(await screen.findByLabelText('transactions')).toBeInTheDocument()

    expect(await screen.findByLabelText('dashboard icon')).not.toHaveStyle(
      `background: ${theme.colors.white}`,
    )
    expect(await screen.findByLabelText('transactions icon')).toHaveStyle(
      `background: ${theme.colors.white}`,
    )
  })
})
