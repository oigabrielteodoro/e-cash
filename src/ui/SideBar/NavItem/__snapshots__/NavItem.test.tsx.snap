// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem should be able match snapshot 1`] = `
.c7 {
  -webkit-transition: 300ms;
  transition: 300ms;
  min-width: 4.2rem;
  min-height: 4.2rem;
  display: grid;
  place-items: center;
  border-radius: 0.5rem;
  position: relative;
  background: #ffffff;
}

.c7 svg {
  -webkit-transition: 300ms;
  transition: 300ms;
  color: #a1a7ae;
}

.c7 svg {
  color: #3a67e1;
}

.c5 {
  -webkit-transition: 300ms;
  transition: 300ms;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
}

.c5:hover .c6 {
  background: #ffffff;
}

.c5:hover .c6 svg {
  color: #3a67e1;
}

.c2 {
  opacity: 0;
  visibility: hidden;
}

.c3 {
  position: absolute;
  color: #18171d;
  font-size: 1.2rem;
  font-weight: 500;
  background: #ffffff;
  padding: 1rem 1.6rem;
  border-radius: 0.5rem;
  white-space: nowrap;
  -webkit-transition: 300ms;
  transition: 300ms;
  border: 0.1rem solid #cfd7e3;
  text-align: center;
  z-index: 1;
  top: 50%;
  right: 0;
  -webkit-transform: translate(calc(100% + 1.6rem),-50%);
  -ms-transform: translate(calc(100% + 1.6rem),-50%);
  transform: translate(calc(100% + 1.6rem),-50%);
}

.c3::before {
  content: '';
  position: absolute;
  border-style: solid;
  border-width: 0.6rem;
  border-color: transparent;
}

.c4 {
  position: absolute;
  background: #ffffff;
  height: 1rem;
  width: 1rem;
  border: 0.1rem solid #cfd7e3;
  z-index: 1;
  top: 50%;
  left: calc(100% + 1.2rem);
  -webkit-transform: translateY(-50%) rotate(-45deg);
  -ms-transform: translateY(-50%) rotate(-45deg);
  transform: translateY(-50%) rotate(-45deg);
  border-bottom-color: transparent;
  border-right-color: transparent;
}

.c0 {
  position: relative;
}

.c0:hover .c1 {
  opacity: 1;
  visibility: visible;
}

<div>
  <li
    class="c0"
  >
    <div
      class="c1 c2"
    >
      <span
        class="c3"
      >
        Link
      </span>
      <div
        class="c4"
      />
    </div>
    <a
      aria-label="link"
      class="c5"
      href="/"
    >
      <div
        aria-label="link icon"
        class="c6 c7"
      >
        <svg
          fill="currentColor"
          height="24"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M924.8 385.6a446.7 446.7 0 0 0-96-142.4 446.7 446.7 0 0 0-142.4-96C631.1 123.8 572.5 112 512 112s-119.1 11.8-174.4 35.2a446.7 446.7 0 0 0-142.4 96 446.7 446.7 0 0 0-96 142.4C75.8 440.9 64 499.5 64 560c0 132.7 58.3 257.7 159.9 343.1l1.7 1.4c5.8 4.8 13.1 7.5 20.6 7.5h531.7c7.5 0 14.8-2.7 20.6-7.5l1.7-1.4C901.7 817.7 960 692.7 960 560c0-60.5-11.9-119.1-35.2-174.4zM761.4 836H262.6A371.12 371.12 0 0 1 140 560c0-99.4 38.7-192.8 109-263 70.3-70.3 163.7-109 263-109 99.4 0 192.8 38.7 263 109 70.3 70.3 109 163.7 109 263 0 105.6-44.5 205.5-122.6 276zM623.5 421.5a8.03 8.03 0 0 0-11.3 0L527.7 506c-18.7-5-39.4-.2-54.1 14.5a55.95 55.95 0 0 0 0 79.2 55.95 55.95 0 0 0 79.2 0 55.87 55.87 0 0 0 14.5-54.1l84.5-84.5c3.1-3.1 3.1-8.2 0-11.3l-28.3-28.3zM490 320h44c4.4 0 8-3.6 8-8v-80c0-4.4-3.6-8-8-8h-44c-4.4 0-8 3.6-8 8v80c0 4.4 3.6 8 8 8zm260 218v44c0 4.4 3.6 8 8 8h80c4.4 0 8-3.6 8-8v-44c0-4.4-3.6-8-8-8h-80c-4.4 0-8 3.6-8 8zm12.7-197.2l-31.1-31.1a8.03 8.03 0 0 0-11.3 0l-56.6 56.6a8.03 8.03 0 0 0 0 11.3l31.1 31.1c3.1 3.1 8.2 3.1 11.3 0l56.6-56.6c3.1-3.1 3.1-8.2 0-11.3zm-458.6-31.1a8.03 8.03 0 0 0-11.3 0l-31.1 31.1a8.03 8.03 0 0 0 0 11.3l56.6 56.6c3.1 3.1 8.2 3.1 11.3 0l31.1-31.1c3.1-3.1 3.1-8.2 0-11.3l-56.6-56.6zM262 530h-80c-4.4 0-8 3.6-8 8v44c0 4.4 3.6 8 8 8h80c4.4 0 8-3.6 8-8v-44c0-4.4-3.6-8-8-8z"
          />
        </svg>
      </div>
    </a>
  </li>
  <div
    class="Toastify"
  />
</div>
`;
